aca_required_temp.py

```diff
@@ -0,0 +1,189 @@
+    return ok_cone_stars
+
+
+def select_fov_stars(ra, dec, roll, cone_stars):
+    edgepad = EDGE_DIST / 5.
+    q = Quat((ra, dec, roll))
+    yag, zag = radec2yagzag(cone_stars['RA_PMCORR'], cone_stars['DEC_PMCORR'], q)
+    row, col = chandra_aca.yagzag_to_pixels(yag * 3600,
+                                            zag * 3600, allow_bad=True)
+    stars_in_fov = cone_stars[(row < (512 - edgepad))
+                              & (row > (-512 + edgepad))
+                              & (col < (512 - edgepad))
+                              & (col > (-512 + edgepad))]
+    return stars_in_fov
+
+
+def max_temp(ra, dec, roll, time, cone_stars):
+    fov_stars = select_fov_stars(ra, dec, roll, cone_stars)
+    if not len(fov_stars):
+        return None
+    # take the 8 brightest
+    stars = fov_stars[0:8]
+    id_hash = tuple(stars['AGASC_ID'])
+    if id_hash in TEMP_CACHE:
+        t_ccd, n_acq = TEMP_CACHE[id_hash]
+    else:
+        t_ccd, n_acq = t_ccd_warm_limit(date=time,
+                                        mags=stars['MAG_ACA'],
+                                        colors=stars['COLOR1'],
+                                        min_n_acq=N_ACQ_STARS,
+                                        cold_t_ccd=COLD_T_CCD,
+                                        warm_t_ccd=WARM_T_CCD)
+        TEMP_CACHE[id_hash] = (t_ccd, n_acq)
+    return t_ccd
+
+
+def get_rolldev(pitch):
+    if pitch < ROLL_TABLE['pitch'][0]:
+        return 0
+    idx = np.searchsorted(ROLL_TABLE['pitch'], pitch)
+    return ROLL_TABLE['rolldev'][idx - 1]
+
+
+def best_temp_roll(ra, dec, nom_roll, day_pitch, time, cone_stars):
+    """
+    Loop over possible roll range for this pitch and return best temp/roll
+    combination
+    """
+    best_t_ccd = None
+    best_roll = None
+    rolldev = get_rolldev(day_pitch)
+    # Quat doesn't care about the domain of roll, so I don't have to
+    # do anything tricky with 180/360 etc
+    for rolldiff in np.arange(0, rolldev, step=.5):
+        # Walk out from nom_roll instead of just through the range,
+        # and quit if we get one that is at the warm limit
+        # That will give us the closest-to-nominal best choice
+        for roll in [nom_roll - rolldiff, nom_roll + rolldiff]:
+            roll_t_ccd = max_temp(ra, dec, roll, time=time, cone_stars=cone_stars)
+            if roll_t_ccd is not None:
+                if best_t_ccd is None or roll_t_ccd > best_t_ccd:
+                    best_t_ccd = roll_t_ccd
+                    best_roll = roll
+                if best_t_ccd == WARM_T_CCD:
+                    break
+    return best_t_ccd, best_roll
+
+
+def temps_for_attitude(ra, dec, start='2014-09-01', stop='2015-12-31'):
+    # reset the caches at every new attitude
+    global TEMP_CACHE
+    TEMP_CACHE.clear()
+
+    # set the agasc lookup time to be in the middle of the cycle for
+    # proper motion correction
+    agasc_mid_time = DateTime((DateTime(start).secs + DateTime(stop).secs)
```

Comment:
What is this the "mid time" of?  It's not the AGASC (which doesn't have a formal time, except that by chance almost all have epoch=J2000.  It's the "mid time" of the LTS being planned.  So:
```
# Center time of the long-term schedule (LTS) cycle ...
lts_mid_time = start + (stop - start) / 2
```
(Again using DateTime arithmetic).